socs/agents/hwp_pmx/agent.py

```diff
@@ -250,6 +285,83 @@ def _stop_acq(self, session, params=None):
         else:
             return False, 'acq is not currently running'
 
+    def initiate_shutdown(self, session, params):
+        """
+        Initiate the shutdown of the agent.
+        """
+        self.log.warn("INITIATING SHUTDOWN")
+
+        with self.lock.acquire_timeout(10, job='shutdown') as acquired:
+            if not acquired:
+                self.log.error("Could not acquire lock for shutdown.")
+                return False, "Could not acquire lock."
+
+            self.shutdown_mode = True
+            self.dev.ign_external_voltage()
+            self.dev.stop()
```

Comment:
I don't see a "stop()" method for self.dev as defined in "/socs/agents/hwp_pmx/drivers/PMX_ethernet.py". If you're intending to turn off the PMX output I would use "self.dev.turn_off()" instead.

Reply:
Ah great, thanks for catching this!